app/(chat)/api/chat/route.ts

```diff
@@ -0,0 +1,94 @@
+import { convertToCoreMessages, streamText } from "ai";
+import { z } from "zod";
+
+import { customModel } from "@/ai";
+import { auth } from "@/app/(auth)/auth";
+import { deleteChatById, getChatById, saveChat } from "@/db/queries";
+
+export async function POST(request: Request) {
+  const { id, messages, selectedFilePathnames } = await request.json();
```

Comment:
we get a lot of request about strongly typing these, any thoughts?

Reply:
added!